test/infrastructure/vcsim/README.md

```diff
@@ -1,4 +1,156 @@
+
+### Preparing up a test environment (using vcsim)
+
+In order to understand the architecture of the vcsim controller, it is convenient to start looking only at 
+components that are involved in setting up a test environment that will use vcsim as a target infrastructure.
+
+A test environment for CAPV requires two main elements:
+
+- A vCenter, or in this case a vcsim instance simulating a vCenter.
+- A VIP for the control plane endpoint
+
+In order to create a vcsim instance it is possible to use the `VCenterSimulator` resource, and the corresponding
+VCenterSimulatorReconciler will take care of the provisioning process.
+
+Note: The vcsim instance will run inside the Pod that host the vcsim controller, and be accessible trough a port that
+will surface in the status of the `VCenterSimulator` resource.
+
+Note: As of today, given a limitation in the vcsim library, only a single instance vcsim instance can be active at any
+time.
+
+In order to get a VIP for the control plane endpoint it is possible to use the `ControlPlaneEndpoint` resource, 
+and the corresponding ControlPlaneEndpointReconciler will take care of the provisioning process.
+
+Note: The code for CAPI's in memory provider is used to create VIPs; they ar implemented as a listener on a port
+in the range 20000-24000 on the vcsim controller Pod. The port will surface in the status of the `ControlPlaneEndpoint`
+resource.
+
+![Architecture](architecture-part1.drawio.svg)
+
+The vcsim controller also implement two additional CRDs:
+
+- The `EnvSubst` CRD, that can be used to generate EnvSubst variables to be used with cluster templates.
+  Each `EnvSubst` generates variables for a single workload cluster, using the VIP from a `ControlPlaneEndpoint` resource
+  and targeting a vCenter that can be originated by a `VCenterSimulator`. 
+
+- The `VMOperatorDependencies` CRD, that is explained in the [vm-operator](../vm-operator/README.md) documentation.
+
+Please also note that the [vcsim.sh](scripts/vcsim.sh) script provide a simplified way for creating `VCenterSimulator`,
+`ControlPlaneEndpoint`, a `EnvSubst` referencing both, and finally copy all the variables to a .env file.
+With the env file, that it is finally possible to create a cluster.
+
+```shell
+# vcsim1 is the name of the vcsim instance to be created
+# cluster1 is the name of the workload cluster to be created (and it used also as a name for the ControlPlaneEndpoint resource)
+$ test/infrastructure/vcsim/scripts/vcsim.sh vcsim1 cluster1
+created VCenterSimulator vcsim1
+created ControlPlaneEndpoint cluster1
+envvar.vcsim.infrastructure.cluster.x-k8s.io/cluster1 created
+created EnvVar cluster1
+done!
+GOVC_URL=https://user:pass@127.0.0.1:36401/sdk
+
+source vcsim.env
+
+# After this command completes you have to run the printed source command
+$ source vcsim.env
+
+# Then you are ready to create a workload cluster
+$ cat <your template> | envsubst | kubectl apply -f -
+```
+
+### Cluster provisioning with vcsim
```

Comment:
Should we mention that this explains it based on govmomi mode? (as vcsim is also compatible with vmoperator)